### DISCUSSION TITLE
How do I resolve these error messages when installing?

### DISCUSSION THREAD

**Discussion by mysticbackpacker** [ReagentX/imessage-exporter#140] [Q&A] [answered]
Bear with me as I'm not a coder, just trying to use this great tool! I followed instructions on [this page](https://lifehacker.com/you-can-export-your-entire-imessage-history-1850000632) and after running the line: "cargo install imessage-exporter" I got an xcrun error. The last part of the message is:

 = note: xcrun: error: invalid active developer path (/Library/Developer/CommandLineTools), missing xcrun at: /Library/Developer/CommandLineTools/usr/bin/xcrun
          

error: could not compile `typenum` (build script) due to previous error
error: failed to compile `imessage-exporter v1.4.0`, intermediate artifacts can be found at `/var/folders/6j/11bhw11n659fcrmm08qfzr5r0000gn/T/cargo-installYric9p`

Could someone point me in the right direction as simply as possible (see intro comment!)? How do I resolve this?

**Comment by ReagentX** [comment, marked as answer]
I believe this means you are missing the Xcode command line tools; on MacOS you can install them with `xcode-select --install`.

**Reply by mysticbackpacker** [reply]
Thank you! I really appreciate it :-)
